### PR TITLE
azure-storage-cpp: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
.

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Review by jwillikers — APPROVED** [review]

File: recipes/azure-storage-cpp/all/conanfile.py [hunk 1]
```diff
@@ -40,84 +38,119 @@ def _minimum_cpp_standard(self):
...
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("cpprestsdk/2.10.19", transitive_headers=True, transitive_libs=True)
+        # LibXML2 v2.12+ from CCI is not compatible. Needs to be built with -DLIBXML_OUTPUT_ENABLED:
+        # https://github.com/GNOME/libxml2/commit/5d2dbe79fae5618d0bd2a7e52be9cf98b11d98ad
+        self.requires("libxml2/2.11.5", transitive_headers=True, transitive_libs=True)
```
Comment (RubenRBS) [inline recipes/azure-storage-cpp/all/conanfile.py:67]: I'm guessing this is still the case?

**Review by RubenRBS — COMMENTED** [review]
Hi! Sorry for the delay on this one, found this PR while looking into the linked `azure-sdk-for-cpp` PR, Ithought it would be great to have both now that I'm reviewing it, even if this has been marked as deprecated upstream

File: recipes/azure-storage-cpp/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/azure-storage-cpp/all/conanfile.py]: Does not look like it. Removed the comment.

**Comment by valgur** [comment]
One of the remaining green migration PRs. Another review would be welcome.

**Review by jwillikers — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**